ql/src/java/org/apache/hadoop/hive/ql/QueryState.java

```diff
@@ -88,6 +88,11 @@ public class QueryState {
    */
   private final ReentrantLock resolveConditionalTaskLock = new ReentrantLock(true);
 
+  /**
+   * command type that requires the authorization.
+   */
+  private HiveOperation authorizationCommandType;
+
```

Comment:
The `HiveOperation` class was explicitly introduced for authorization purposes. Adding a separate field in `QueryState` is confusing and will be hard to maintain. It becomes unclear when/how to update `commandType` and `authorizationCommandType`, why we need both, when and under what context we should use each of those.